llvm/lib/Transforms/Vectorize/VPlanConstruction.cpp

```diff
@@ -634,6 +634,48 @@ void VPlanTransforms::createLoopRegions(VPlan &Plan) {
   TopRegion->getEntryBasicBlock()->setName("vector.body");
 }
 
+void VPlanTransforms::removeInvariantStoresOfReduction(
```

Comment:
I am not sure if it is worth moving this out, as long as it is based on LLVM IR references